src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -803,6 +803,14 @@ private HttpRequestMessage CreateRequest<TResult>(ServerNode node, RavenCommand<
                     throw new AuthorizationException("Forbidden access to " + chosenNode.Database + "@" + chosenNode.Url + ", " +
                         (Certificate == null ? "a certificate is required." : Certificate.FriendlyName + " does not have permission to access it or is unknown.") +
                         request.Method + " " + request.RequestUri);
+                case HttpStatusCode.Gone: // request not relevant for the chosen node - the database has been moved to a different one
+                    if (shouldRetry == false)
+                        return false;
+
+                    await UpdateTopologyAsync(chosenNode, Timeout.Infinite, forceUpdate: true).ConfigureAwait(false);
```

Comment:
@ayende 
The reason I had to force topology update here was that both typologies, the previous and just retrieved ones, have `Topology.Etag == -1` so the new one was filtered out here:
https://github.com/ravendb/ravendb/blob/v4.0/src/Raven.Client/Http/NodeSelector.cs#L54